Doctrine/EmailUpdateListener.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace FOS\UserBundle\Doctrine;
+
+use Doctrine\Common\EventSubscriber;
+use Doctrine\Common\Persistence\Event\LifecycleEventArgs;
+use Doctrine\ORM\Event\PreUpdateEventArgs;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use FOS\UserBundle\Model\UserInterface;
+use FOS\UserBundle\Services\EmailConfirmation\EmailUpdateConfirmation;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+
+/**
+ * Class EmailUpdateListener
+ * @package FOS\UserBundle\Doctrine
+ */
+class EmailUpdateListener implements EventSubscriber
+{
+    /**
+     * @var EmailUpdateConfirmation
+     */
+    private $emailUpdateConfirmation;
+
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * Constructor
+     *
+     * @param ContainerInterface $container
+     * @param RequestStack $requestStack
+     */
+    public function __construct(ContainerInterface $container, RequestStack $requestStack)
+    {
+        $this->container = $container;
```

Comment:
It would be better to inject `fos_user.email_update_confirmation` instead of the entire container.